contracts/Registrar.sol

```diff
@@ -6,86 +6,144 @@ import "@ensdomains/ens/contracts/ENS.sol";
+
+    /// @notice A commitment was made
+    event CommitmentMade(bytes32 commitment, uint blockNumber);
+
+    /// @notice The contract admin was changed
+    event AdminChanged(address newAdmin);
+
+    /// @notice The registration fee was changed
+    event RegistrationFeeChanged(uint amt);
+
+    /// @notice The ownership of the domain was changed
+    event DomainOwnershipChanged(address newOwner);
+
+    /// @notice The registration fee was changed
+    event ResolverChanged(address resolver);
+
+    /// @notice The registration fee was changed
+    event TTLChanged(uint64 amt);
+
+    /// @notice The ens registry was updated
+    event EnsChanged(address ens);
+
+    /// @notice The minimum age for a commitment was changed
+    event MinCommitmentAgeChanged(uint256 amt);
+
+    // --- AUTH ---
+
+    /// The contract admin who can set fees.
+    address public admin;
 
     /// Protects admin-only functions.
     modifier adminOnly {
-        require(msg.sender == admin, "Only the admin can perform this action");
+        require(msg.sender == admin, "Registrar: only the admin can perform this action");
         _;
     }
 
+    // --- INIT ---
+
     constructor(
         ENS _ens,
-        bytes32 ethDomainNameHash,
-        uint256 ethDomainTokenId,
-        address _oracleAddress,
-        address _exchangeAddress,
         ERC20Burnable _rad,
-        address adminAddress
+        address _admin
     ) {
         ens = _ens;
-        oracleAddress = _oracleAddress;
-        exchangeAddress = _exchangeAddress;
         rad = _rad;
-        domain = ethDomainNameHash;
-        tokenId = ethDomainTokenId;
-        admin = adminAddress;
-    }
-
-    /// Register a subdomain using ether.
-    function registerEth(string memory, address) public payable {
-        revert("Registrar::registerEth: Registration using ETH is not yet available");
+        admin = _admin;
     }
 
-    /// Register a subdomain using radicle tokens.
-    function registerRad(string memory name, address owner) public {
-        uint256 fee = registrationFeeRad;
-
-        require(rad.balanceOf(msg.sender) >= fee, "Transaction includes registration fee");
-        require(valid(name), "Name must be valid");
+    // --- USER FACING METHODS ---
 
-        _register(keccak256(bytes(name)), owner);
+    /// Commit to a future name registration
+    function commit(bytes32 commitment) external {
+        require(commitments.commited(commitment) == 0, "Registrar::commit: already commited");
+        require(rad.balanceOf(msg.sender) >= fee, "Registrar::register: insufficient rad balance");
 
         rad.burnFrom(msg.sender, fee);
+        commitments.commit(commitment);
+
+        emit CommitmentMade(commitment, block.number);
     }
 
-    function _register(bytes32 label, address owner) private {
-        bytes32 node = namehash(domain, label);
+    /// Register a subdomain (with a custom resolver and ttl)
```

Comment:
So this function an be called by anyone. That could be useful if we want to bare the costs.